e2e/autoscript-apps/src/authn-protect-metadata/autoscript.ts

```diff
@@ -0,0 +1,108 @@
+/*
+ * @forgerock/javascript-sdk
+ *
+ * autoscript.ts
+ *
+ * Copyright (c) 2024 ForgeRock. All rights reserved.
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+// @ts-nocheck
+import * as forgerock from '@forgerock/javascript-sdk';
+import { PIProtect } from '@forgerock/ping-protect';
+import { delay as rxDelay, map, mergeMap } from 'rxjs/operators';
+import { from } from 'rxjs';
+
+function autoscript() {
+  const delay = 0;
+
+  const url = new URL(window.location.href);
+  const amUrl = url.searchParams.get('amUrl') || 'http://localhost:9443/am';
+  const realmPath = url.searchParams.get('realmPath') || 'root';
+  const un = url.searchParams.get('un') || 'sdkuser';
+  const pw = url.searchParams.get('pw') || 'password';
+  const pauseBehaviorData = url.searchParams.get('pauseBehaviorData') || 'true';
+  const tree = url.searchParams.get('tree') || 'TEST_MetadataMarketPlace';
+
+  console.log('Configure the SDK');
+  forgerock.Config.set({
+    realmPath,
+    tree,
+    serverConfig: {
+      baseUrl: amUrl,
+    },
+  });
+
+  try {
+    forgerock.SessionManager.logout();
+  } catch (err) {
+    // Do nothing
+  }
+
+  console.log('Initiate first step with `undefined`');
+  // Wrapping in setTimeout to give the test time to bind listener to console.log
+  setTimeout(function () {
+    from(forgerock.FRAuth.start())
+      .pipe(
+        mergeMap(
+          (step) => {
+            try {
+              const config = PIProtect.getNodeConfig(step);
+
+              console.log('start protect');
+              return PIProtect.start(config);
+            } catch (err) {
+              const cb = (step as forgerock.Step).getCallbackOfType(
+                'MetadataCallback',
+              ) as forgerock.MetadataCallback;
+
+              clientError.setInputValue('we had a failure');
```

Comment:
Where does `clientError` come from? I'm assuming this is from the original implementation? We can change this to use the main Protect module, rather than using callbacks, yeah?